db.py

```diff
@@ -42,34 +41,56 @@ def addView(self, connection, url):
 
     def getTopSites(self, connection, amount=10):
         """ Get the top domains using this tool by hits. Ignore specified domains """
+        top_urls = self.getTopUrls(connection, -1)
+
+        # Get total hits per domain
+        site_counts = defaultdict(int)
+        for url in top_urls['urls']:
+            # Get the domain - part before the first '/'
+            domain = url.split('/')[0]
+            # Add hit counts to the domain
+            site_counts[domain] += top_urls['values'][url]
+
+        # Sort the domains by hits
+        sorted_sites = sorted(site_counts, key=lambda x: site_counts[x], reverse=True)
+
+        # Return sorted domains and their values, this allows for lower Python version support
+        return {
+            'domains': sorted_sites[:amount],
+            'values': {site: site_counts[site] for site in site_counts}
+        }
+
+    def getTopUrls(self, connection, amount=10):
+        """ Get the top urls using this tool by hits. Ignore specified domains """
         # Select all urls and counts
+        if amount < 0:
```

Comment:
`None` would probably be a better option to use rather than using 0 and limiting to `2147483647`

Reply:
I found that `-1` is actually the right value to use to not limit the query at all: https://stackoverflow.com/a/19676495/3112139. But I'm gonna revert the query anyway, see below.